clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3180,6 +3182,7 @@ private int processKafkaDataMessage(
                 ByteUtils.extractByteArray(put.getPutValue()), // must be done before recordTransformer changes putValue
                 AvroProtocolDefinition.CHUNKED_VALUE_MANIFEST.getCurrentProtocolVersion());
             hostLevelIngestionStats.recordAssembledValueSize(chunkedValueManifest.getSize(), currentTimeMs);
+            recordAssembledRecordSizeRatio(keyLen + chunkedValueManifest.getSize(), currentTimeMs); // TODO: RMD?
```

Comment:
`RMD` is a good concern, and maybe we should emit two metrics: one for user supplied value and another one is for `RMD`, and when we setup alert, we should only set it up against the value size ratio and for RMD, we only need to emit the absolute size, no ratio is needed and for now, it will be mostly for informational purpose.

Reply:
That would be an nice idea, but I do not know how to determine whether it is RMD or not. As far as I can tell from `WritingChunkingHelper#chunkPayloadAndSend()`, the `ChunkedValueManifest` is the same for both RMD and value?

I spoke with @sixpluszero and due to the difficulty in keeping track of value manifest (which only has value size) and rmd manifest (for chunked rmd) together, it might be better to leave it as-is and recognize this limitation in the calculation of size. Since we control the threshold for alerting customers, perhaps we could make it lower to account for the missing RMD?

Reply:
I think we need to differentiate the chunked payload is for value or RMD, and for RMD, we shouldn't alert the customers as RMD is the internal component to handle DCR and customers shouldn't need to be aware of that, and also today in theory, RMD can grow to be very large as it is keeping tombstones.

Reply:
I checked with @sixpluszero and I have a method of differentiating whether a manifest pertains to a value or RMD, so I should be able to separate the two now.

Reply:
I do find it a bit odd that the alert on the customer side does not include RMD size, but [the condition for `RecordTooLarge` does include RMD size in being `key + value + rmd`](https://github.com/linkedin/venice/blob/4a20cdf03d362989bfcdf0ee22ffd7f2038fe50a/internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java#L909)? If the RMD size is very large, then maybe we could see `RecordTooLarge` failures before the ratio reaches high levels?

Reply:
I think we should mutate the logic in VeniceWriter, so that the limit only applies to the value record and once we have a separate metric to track the RMD size, it will be clear which part is causing issue.
RMD is the internal component, which Venice team is responsible for the size optimization, and we shouldn't fail user's write if RMD is too large from my understanding.

Reply:
Even we want to put up a limit against RMD size to avoid the damage to other tenants, we should use a separate feature flag to control this logic and it should be off by default and we can use an uniform limit controlled by Venice Config from the beginning.

Reply:
Understood. I'll change the logic in `VeniceWriter` to only look at `key + value` size:

```java
int veniceRecordSize = serializedKey.length + serializedValue.length + replicationMetadataPayloadSize;
if (isChunkingNeededForRecord(veniceRecordSize)) { // ~1MB default
  // RMD size is not checked because it's an internal component, and a user's write should not be failed due to it
  if (isChunkingEnabled && !isRecordTooLarge(serializedKey.length + serializedValue.length)) {
```

I think RMD is also stored in a separate column, so maybe it'll be fine not to count it at all?

Reply:
RMD size should be dealt with separately as it is invisible to the end user and if we alert it, it would cause a lot of confusions.

Reply:
Does this means we will eventually set the threshold based on pure user value size, and internally given that we know RMD size is kind of proportional to value size, the actual limit would be as much as 2x of the set up threshold?